app/views/admin/enterprises/_form.html.haml

```diff
@@ -1,63 +1,21 @@
-
-%fieldset.alpha.no-border-bottom{ ng: { show: "menu.selected.name=='about'" } }
-  %legend {{menu.selected.label}}
-  = render 'admin/enterprises/form/about_us', f: f
-
-%fieldset.alpha.no-border-bottom{ ng: { show: "menu.selected.name=='images'" } }
-  %legend {{menu.selected.label}}
-  = render 'admin/enterprises/form/images', f: f
-
-%fieldset.alpha.no-border-bottom{ ng: { show: "menu.selected.name=='properties'" } }
-  %legend {{menu.selected.label}}
-  = render 'admin/enterprises/form/properties', f: f
-
-%fieldset.alpha.no-border-bottom{ ng: { show: "menu.selected.name=='shipping_methods'" } }
-  %legend {{menu.selected.label}}
-  = render 'admin/enterprises/form/shipping_methods', f: f
-
-%fieldset.alpha.no-border-bottom{ ng: { show: "menu.selected.name=='payment_methods'" } }
-  %legend {{menu.selected.label}}
-  = render 'admin/enterprises/form/payment_methods', f: f
-
-%fieldset.alpha.no-border-bottom{ ng: { show: "menu.selected.name=='enterprise_fees'" } }
-  %legend {{menu.selected.label}}
-  = render 'admin/enterprises/form/enterprise_fees', f: f
-
-%fieldset.alpha.no-border-bottom{ ng: { show: "menu.selected.name=='enterprise_permissions'" } }
-  %legend {{menu.selected.label}}
-
-%fieldset.alpha.no-border-bottom{ ng: { show: "menu.selected.name=='inventory_settings'" } }
-  %legend {{menu.selected.label}}
-  = render 'admin/enterprises/form/inventory_settings', f: f
-
-%fieldset.alpha.no-border-bottom{ ng: { show: "menu.selected.name=='shop_preferences'" } }
-  %legend {{menu.selected.label}}
-  = render 'admin/enterprises/form/shop_preferences', f: f
-
-%fieldset.alpha.no-border-bottom{ ng: { if: "menu.selected.name=='tag_rules'" } }
-  %legend {{menu.selected.label}}
-  = render 'admin/enterprises/form/tag_rules', f: f
+- enterprise_side_menu_items(@enterprise).each do |item|
+  - case item[:name]
+  - when 'primary_details'
+    %fieldset.alpha.no-border-bottom{ id: "#{item[:name]}_panel", data: { controller: "primary-details", "primary-details-primary-producer-value": @enterprise.is_primary_producer.to_s, "primary-details-enterprise-sells-value": @enterprise.sells, "tabs-and-panels-target": "panel default" }}
+      %legend= t("#{ item[:name] }")
+      = render "admin/enterprises/form/#{ item[:form_name] || item[:name] }", f: f
+
+  - when 'address'
+    = f.fields_for :address do |af|
+      %fieldset.alpha.no-border-bottom{ id: "#{item[:name]}_panel", data: { "tabs-and-panels-target": "panel" }}
+        %legend= t("#{ item[:name] }")
+        = render 'admin/enterprises/form/address', af: af
+
+  - when 'enterprise_permissions'
+    %fieldset.alpha.no-border-bottom{ id: "#{item[:name]}_panel", data: { "tabs-and-panels-target": "panel" }}
+      %legend= t("#{ item[:name] }")
+
+  - else
+    %fieldset.alpha.no-border-bottom{ id: "#{item[:name]}_panel", data: { "tabs-and-panels-target": "panel"  }}
+      %legend= t("#{ item[:name] }")
+      = render "admin/enterprises/form/#{ item[:form_name] || item[:name] }", f: f
```

Comment:
Can we condense this? I'm thinking that we could add more data to the items, like additional controllers, a flag for `fields_for` and a flag for no partial rendering. Maybe another PR?

Reply:
I'm sorry, I don't understand what `condensing` means here...Also, you talk about flags but I have not flags in rails :sweat_smile:

Reply:
I meant to make it even shorter, get rid of the `case` statement and build all the flexibility we need into the one fieldset template. But don't worry, it was just an idea. We can leave it as is.